platform/default/mbgl/map/map_snapshotter.hpp

```diff
@@ -0,0 +1,50 @@
+#pragma once
+
+#include <mbgl/util/image.hpp>
+#include <mbgl/util/thread.hpp>
+#include <mbgl/util/optional.hpp>
+
+#include <exception>
+#include <string>
+#include <functional>
+
+namespace mbgl {
+
+template<class> class ActorRef;
+struct CameraOptions;
+class FileSource;
+class Size;
+class LatLngBounds;
+
+class SnapshotCallback {
```

Comment:
You might be able to do `using SnapshotCallback = std::function<void (std::exception_ptr, PremultipliedImage)>` and avoid the wrapper class.